modules/sievefilters/modules.php

```diff
@@ -1545,9 +1545,9 @@ class Hm_Sieve_Client_Factory {
     public function init($user_config = null, $imap_account = null)
     {
         if ($imap_account && ! empty($imap_account['sieve_config_host'])) {
-            $sieve_options = explode(':', $imap_account['sieve_config_host']);
-            $client = new PhpSieveManager\ManageSieve\Client($sieve_options[0], $sieve_options[1]);
-            $client->connect($imap_account['user'], $imap_account['pass'], false, "", "PLAIN");
+            list($sieve_hostname, $sieve_port, $sieve_tls) = parse_sieve_config_host($imap_account['sieve_config_host']);
+            $client = new PhpSieveManager\ManageSieve\Client($sieve_hostname, $sieve_port);
+            $client->connect($imap_account['user'], $imap_account['pass'], $sieve_tls, "", "PLAIN");
```

Comment:
Add try/catch block here.

Reply:
@kroky changes have been made. Thanks.

Reply:
I noticed that the tls option in $client->connect function was not taking in account in henrique-borba/sieve-manager. I will do a merge request for it.